code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -388,6 +388,8 @@
 	taste_description = "mushroom"
 
 /datum/reagent/drug/mushroomhallucinogen/on_mob_life(mob/living/carbon/M)
+	if(ispsyphoza(M))
```

Comment:
you could probably add a new mobtype here (like INSECT, for MUSHROOM that makes it immune to mushroom stuff)

Reply:
Nah